lua52-sys/src/lib.rs

```diff
@@ -27,6 +27,7 @@ pub const LUA_ERRERR: c_int = 6;
 pub struct lua_State;
 
 pub type lua_CFunction = extern "C" fn(L: *mut lua_State) -> c_int;
+pub type lua_CFunction2 = unsafe extern "C" fn(*mut lua_State);
```

Comment:
You should change `open_helper` to accept a `unsafe extern "C" fn(*mut lua_State)` instead of a `ffi::lua_CFunction2` (which allows you to remove `lua_CFunction2` from the bindings).

Then in `open_helper` you should be able to do something like `func as ffi::lua_CFunction`. If it doesn't work, use `mem::transmute(func)`. It's the last resort solution, but at least it would compile.